### ISSUE TITLE
How does "Replace Host" rule work?

### ISSUE THREAD

**Issue by Plushfire5** [requestly/customer-support#64]
I've read the documentation ( https://docs.requestly.io/using-rules/replace-rule ) but I don't quite understand it, so I would like to ask if somebody can help me with my case.

I have this URL

`https://ext.packagetrack.com/api/helper.php?carrier=(*1)&trackingnumber=(*2)`

which when opened, I want it to redirect to this one

`https://t.17track.net/es#nums=(*3)`

where:
(*1) would change everytime, so I need it to ignore that string of the URL
(*2) needs to be placed in (*3)'s place

Example:
Turn this
`https://ext.packagetrack.com/api/helper.php?carrier=USPS&trackingnumber=AA12345678BB`
Into this
`https://t.17track.net/es#nums=AA12345678BB`

Basically I want the generated URL by a chrome extension (PackageTrack) to turn into a URL for 17track.net but keeping the shipping code.

How can I do this?

**Comment by requestly-team** [comment]
Hi @Plushfire5, 

There are multiple ways to achieve your use case. Let`s discuss the simplest one.

First create a Modify Query Parameter Rule to remove parameter "carrier" (since it would change every time and isn't required in our target URL)

Request `URL` `Contains` `https://ext.packagetrack.com/api/helper.php?carrier=`
`Remove` Param `carrier`

Then, create a Replace Host Rule to replace parts of our URL

Replace `https://ext.packagetrack.com/api/helper.php?trackingnumber=`  With `https://t.17track.net/es#nums=`

Request `URL` `Contains` `https://ext.packagetrack.com/api/helper.php?trackingnumber=`

**Outcome:**

Input URL: `https://ext.packagetrack.com/api/helper.php?carrier=USPS&trackingnumber=AA12345678BB`

After Modify Query Parameter Rule: `https://ext.packagetrack.com/api/helper.php?trackingnumber=AA12345678BB`

After Replace Host Rule: `https://t.17track.net/es#nums=AA12345678BB`

which matches the requirement.

**Shortcut to create these rules:**

Create a text file (.txt) with following content:

`[
  {
    "creationDate": 1591717943253,
    "description": "Replace parts in URL to redirect PackageTrack to 17Track",
    "groupId": "",
    "id": "Replace_1591717943253",
    "isSample": false,
    "name": "PackageTrack to 17Track",
    "objectType": "rule",
    "pairs": [
      {
        "from": "https://ext.packagetrack.com/api/helper.php?trackingnumber=",
        "source": {
          "filters": {},
          "key": "Url",
          "operator": "Contains",
          "value": "https://ext.packagetrack.com/api/helper.php?trackingnumber="
        },
        "status": "Inactive",
        "to": "https://t.17track.net/es#nums="
      }
    ],
    "ruleType": "Replace",
    "status": "Active",
    "version": 2
  },
  {
    "creationDate": 1591717823701,
    "description": "Remove query parameter \"carrier\" on ext.packagetrack.com",
    "groupId": "",
    "id": "QueryParam_1591717823701",
    "isSample": false,
    "name": "PackageTrack - Remove QP",
    "objectType": "rule",
    "pairs": [
      {
        "modifications": [
          {
            "actionWhenParamExists": "Overwrite",
            "param": "carrier",
            "type": "Remove",
            "value": ""
          }
        ],
        "source": {
          "filters": {},
          "key": "Url",
          "operator": "Contains",
          "value": "https://ext.packagetrack.com/api/helper.php?carrier="
        }
      }
    ],
    "ruleType": "QueryParam",
    "status": "Active"
  }
]`

Import this file to your Requestly Dashboard by clicking Upload Rule link in the top right dropdown.

**Tip**: Mark these Rules as favourite and quickly toggle them using the Requestly icon on your address bar.

Please let us know if this works for you.

Thanks,
Requestly Team

**Comment by Plushfire5** [comment]
> Hi @Plushfire5,
> 
> There are multiple ways to achieve your use case. Let`s discuss the simplest one.
> 
> First create a Modify Query Parameter Rule to remove parameter "carrier" (since it would change every time and isn't required in our target URL)
> 
> Request `URL` `Contains` `https://ext.packagetrack.com/api/helper.php?carrier=`
> `Remove` Param `carrier`
> 
> Then, create a Replace Host Rule to replace parts of our URL
> 
> Replace `https://ext.packagetrack.com/api/helper.php?trackingnumber=` With `https://t.17track.net/es#nums=`
> 
> Request `URL` `Contains` `https://ext.packagetrack.com/api/helper.php?trackingnumber=`
> 
> **Outcome:**
> 
> Input URL: `https://ext.packagetrack.com/api/helper.php?carrier=USPS&trackingnumber=AA12345678BB`
> 
> After Modify Query Parameter Rule: `https://ext.packagetrack.com/api/helper.php?trackingnumber=AA12345678BB`
> 
> After Replace Host Rule: `https://t.17track.net/es#nums=AA12345678BB`
> 
> which matches the requirement.
> 
> **Shortcut to create these rules:**
> 
> Create a text file (.txt) with following content:
> 
> `[ { "creationDate": 1591717943253, "description": "Replace parts in URL to redirect PackageTrack to 17Track", "groupId": "", "id": "Replace_1591717943253", "isSample": false, "name": "PackageTrack to 17Track", "objectType": "rule", "pairs": [ { "from": "https://ext.packagetrack.com/api/helper.php?trackingnumber=", "source": { "filters": {}, "key": "Url", "operator": "Contains", "value": "https://ext.packagetrack.com/api/helper.php?trackingnumber=" }, "status": "Inactive", "to": "https://t.17track.net/es#nums=" } ], "ruleType": "Replace", "status": "Active", "version": 2 }, { "creationDate": 1591717823701, "description": "Remove query parameter \"carrier\" on ext.packagetrack.com", "groupId": "", "id": "QueryParam_1591717823701", "isSample": false, "name": "PackageTrack - Remove QP", "objectType": "rule", "pairs": [ { "modifications": [ { "actionWhenParamExists": "Overwrite", "param": "carrier", "type": "Remove", "value": "" } ], "source": { "filters": {}, "key": "Url", "operator": "Contains", "value": "https://ext.packagetrack.com/api/helper.php?carrier=" } } ], "ruleType": "QueryParam", "status": "Active" } ]`
> 
> Import this file to your Requestly Dashboard by clicking Upload Rule link in the top right dropdown.
> 
> **Tip**: Mark these Rules as favourite and quickly toggle them using the Requestly icon on your address bar.
> 
> Please let us know if this works for you.
> 
> Thanks,
> Requestly Team

Hello,

This works, altough not exactly for my case, due to how the Chrome extension works. If I open a link with the carrier parameter, it will be removed, but then the extension will go to a different page of its own, but if I copy and paste a link with the carrier parameter already removed, it will go directly to 17track. This issue is not related to Requestly though, so I'll see if I can find another way to track my shippments now that I understand how requestly works a bit better. Thanks!
